File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    
    var mini=prices[0];
    
    var n=prices.length;
    
    var profit=0;
    
    for(var i=0;i<n;i++)
        {
            mini=Math.min(mini,prices[i]);
            
            profit=Math.max(profit,prices[i]-mini);
        }
    
    return profit;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int n=prices.size();
        
        int mini=prices[0];
        
        int maxi=INT_MIN;
        
        for(int i=0;i<n;i++)
        {
            mini=min(mini,prices[i]);
            
            maxi=max(maxi,prices[i]-mini);
        }
        
        if(maxi==INT_MIN)return 0;
        return maxi;
        
    }
};